engine/src/main/java/graphql/nadel/engine/execution/NadelExecutionStrategy.java

```diff
@@ -227,9 +228,22 @@ public NadelExecutionStrategy(List<Service> services,
             Service service = entry.getKey();
             Set<GraphQLFieldDefinition> secondLevelFieldDefinitionsForService = entry.getValue();
 
-            Optional<MergedField> maybeNewMergedField = MergedFieldUtil.includeSubSelection(mergedField, namespacedObjectType, executionCtx,
-                    field -> secondLevelFieldDefinitionsForService.stream()
-                            .anyMatch(graphQLFieldDefinition -> graphQLFieldDefinition.getName().equals(field.getName()) || field.getName().equals(Introspection.TypeNameMetaFieldDef.getName())));
+            final boolean serviceExtendsNamespaceType = service.getDefinitionRegistry().getDefinitions(ObjectTypeExtensionDefinition.class)
+                    .stream()
+                    .anyMatch(objectTypeDef -> objectTypeDef.getName().equals(namespacedObjectType.getName()));
+
+            Optional<MergedField> maybeNewMergedField = MergedFieldUtil.includeSubSelection(
+                    mergedField,
+                    namespacedObjectType,
+                    executionCtx,
+                    field -> secondLevelFieldDefinitionsForService
```

Comment:
stretching for a nit here but I think it would help to improve readability a bit if we had 2 helper functions:
`boolean isFieldIncluded(..)` and
`boolean serviceOwnsNamespacedType(..)`